macos/FluentUI/AvatarView/AvatarView.swift

```diff
@@ -409,22 +429,34 @@ open class AvatarView: NSView {
 		return initials?.localizedUppercase
 	}
 
-	/// Returns a color table index for a given display name
-	/// - parameter identifyingString: the unique identifying string for the user
-	///
-	/// - Returns: a unique index generated by walking backwards through the letters in the \p identifyingString
-	/// creating a cumulative hash
-	///
-	/// - note: the returned index is not limited to any size and should not be considered safe for indexing into an array without
-	/// first checking the size of the array in question
-	@objc(colorIndexForIdentifyingString:)
-	public static func colorIndex(for identifyingString: String) -> Int {
-		return identifyingString.utf16.enumerated().reversed().reduce(0) { (hashCode, enumeratedCodePoints) -> Int in
-			// upcast our UInt16s to standard Ints as that's how the hashing algorithm works in other codebases
-			let integerCodePoint = Int(enumeratedCodePoints.element)
-			let shift = enumeratedCodePoints.offset % 8
-			return hashCode ^ Int((integerCodePoint << shift) + (integerCodePoint >> (8 - shift)))
+	@objc(getInitialsColorSetFromPrimaryText:secondaryText:)
+	public static func getInitialsColorSet(fromPrimaryText primaryText: String?, secondaryText: String?) -> ColorSet {
+		// Set the color based on the primary text and secondary text
+		var combined: String
+		if let secondaryText = secondaryText, let primaryText = primaryText, secondaryText.count > 0 {
+			combined = primaryText + secondaryText
+		} else if let primaryText = primaryText {
+			combined = primaryText
+		} else {
+			combined = ""
+		}
+
+		let colors = AvatarView.avatarColors
+		let combinedHashable = combined as NSString
+		let hashCode = Int(abs(javaHashCode(combinedHashable)))
+		return colors[hashCode % colors.count]
+	}
+
+	/// To ensure iOS/MacOS and Android achieve the same result when generating string hash codes (e.g. to determine avatar colors) we've copied Java's String implementation of `hashCode`.
```

Comment:
we should take one step further and actually have a .swift file that is shared between ios and mac projects

Reply:
we have a lot of scope to share code across both repos, the handling of pod spec and framework is beyond the scope of this PR. This PR is required to update devmain code for MacOS Avatar view that I intend to merge by Monday to unblock other PR's

Reply:
Although I understand that refactoring this code and start sharing more across mac/ios is beyond the scope of this task, we really need to start laying out foundation for that.
Do we have a task in ADO to track that?

Reply:
Is there an ADO item to track that?